documentation/trouble_low_catalog_duplication.markdown

```diff
@@ -1,172 +0,0 @@
----
-title: "PuppetDB 3.1 » Troubleshooting » Low Catalog Duplication"
-layout: default
-canonical: "/puppetdb/latest/trouble_low_catalog_duplication"
----
-
-[maintaining_tuning]: ./maintain_and_tune.html
-[configure_vardir]: ./configure.html#vardir
-[global_config]: ./configure.html#global-settings
-[configure_vardir]: ./configure.html#catalog-hash-conflict-debugging
-[configure_catalog_debugging]: ./configure.html#catalog-hash-conflict-debugging
-
-What is Catalog Duplication?
------
-
-Every puppet agent run on a host results in a new catalog sent from
-the puppet master to PuppetDB. Although catalogs can be very large,
-typically there are 1000 or more resources and their relationships in
-each of the catalogs. Storing each of these resources and
-relationships is a fairly I/O-intensive operation on the database used
-by PuppetDB.
-
-Puppet agent runs happen frequently; typically every 60
-minutes. Most of these runs will not result in a catalog with
-different resources or different relationships. Typically, only a
-manually initiated change to Puppet content, such as modification of a
-manifest or a Puppet extension, will update a node's catalog.
-
-PuppetDB stores catalogs with this use case in mind. PuppetDB will SHA1 hash
-the contents of a catalog and will compare that hash with the hash of
-the previously received catalog for that host. Only when these hashes
-don't match will PuppetDB incur the extra cost/overhead of storing the
-new catalog (with all of its resources and edges).
-
-The _catalog duplication rate_ is a measure of how often PuppetDB is able
-to _not_ store a whole catalog. It can be found on the PuppetDB dashboard as a
-percentage. (More information on the dashboard can be found in the
-[Maintaining and Tuning guide][maintaining_tuning].) Typically, this
-catalog duplication rate should be over 90%. If this percentage is
-low, there will be a significantly higher I/O load on the database.
-This load can lead to slowness and resource contention for the
-database and PuppetDB itself.
-
-What Causes Catalogs to Hash Differently?
------
-
-Anything that changes, adds or removes a resource or a relationship
-will cause the catalog for that host to hash differently. Most
-commonly, this is caused by an admin initiated change to a host (i.e.
-adding a new package to the configuration of a web server).
-
-Prior to
-PuppetDB 1.6, this could also be caused by a subtle reordering of
-resources or properties. One example of that scenario is in a two
-puppet master setup, one puppet master would send its resources in a
-different order than the other puppet master. This would cause the
-catalogs to hash differently, even though their content was the same.
-This issue has been fixed in PuppetDB 1.6.
-
-How Can I Detect the Cause of Catalogs Hashing Differently?
------
-
-Unfortunately, diagnosing why a catalog is hashing differently can be
```

Comment:
I'd be nice if we could give some kind of SQL query that would be helpful in this circumstance. Down to this point the doc is still relevant I'd say. I'll give it some thought.